### PR TITLE
feat(build): build Envoy main branch on schedule

### PR DESCRIPTION
**Pull request by lukidzi**
* release hosts on schedule(everyday at 10 am)
* build the main envoy branch on Monday 6am
* fixes to build
* split fips from linux/amd to speedup build
* use better linux machine to reduce build time

### CODE REVIEW

File: .github/workflows/release-hosts.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  schedule:
+    - cron: 0 10 * * *
+
+jobs:
+  release_hosts:
+    secrets:
+      AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+    steps:
+      - uses: actions/checkout@v3
+      # TODO use aws ssm
```
Comment (slonka) [inline .github/workflows/release-hosts.yaml:14]: is this to be done in this PR or the next one? why is aws ssm better?

File: scripts/dns_filter_resolver.h.patch [hunk 2]
```diff
@@ -1,30 +0,0 @@
-diff --git a/source/extensions/filters/udp/dns_filter/dns_filter_resolver.h b/source/extensions/filters/udp/dns_filter/dns_filter_resolver.h
```
Comment (slonka) [inline scripts/dns_filter_resolver.h.patch:1]: I'm assuming we no longer need this patch because it was merged in Envoy, right?

**Review by slonka — APPROVED** [review]
In general OK, one todo question

File: scripts/dns_filter_resolver.h.patch (hunk 2 above)
Comment (lukidzi) [reply scripts/dns_filter_resolver.h.patch:1]: Yes, we don't need

File: .github/workflows/release-hosts.yaml (hunk 1 above)
Comment (lukidzi) [reply .github/workflows/release-hosts.yaml:14]: In next step

**Pull request merged**